dandi/download.py

```diff
@@ -1122,19 +1123,36 @@ def _check_if_more_attempts_allowed(
             return None
         elif retry_after := exc.response.headers.get("Retry-After"):
             # playing safe
-            if not str(retry_after).isdigit():
-                # our code is wrong, do not crash but issue warning so
-                # we might get report/fix it up
-                lgr.warning(
-                    "%s - download failed due to response %d with non-integer"
-                    " Retry-After=%r: %s",
-                    path,
-                    exc.response.status_code,
-                    retry_after,
-                    exc,
-                )
-                return None
-            sleep_amount = int(retry_after)
+            retry_after = str(retry_after)
+            if retry_after.isdigit():
+                sleep_amount = int(retry_after)
```

Comment:
If we are assuming that this client always connect to our trusted DANDI API, I think this is fine, but if we want to want to play even "safer" I think we may want to ensure the `retry_after` string is really a non-negative integer in a decimal expression.

If  `retry_after` is `"²³"` will crash this client at this point. 

This safe guard can be excessive though since we have control of both the client and server.

I think use of `str.isdecimal()` and ensuring the return of `int()` to be non-negative is better.